Tests/CommonTests/DiagnosticAssemblyScannerTests.cs

```diff
@@ -22,71 +23,120 @@ public class DiagnosticAssemblyScannerTests
+        [DeploymentItem("ExampleAnalyzer1.dll", "./oneAnalyzer")]
+        public void ScanDirectory_OneAnalyzerAssembly()
         {
             // Arrange
             TestLogger logger = new TestLogger();
             DiagnosticAssemblyScanner scanner = new DiagnosticAssemblyScanner(logger);
 
-            string noAnalyzerAssemblyPath = typeof(DiagnosticAssemblyScannerTests).Assembly.Location;
-            string testDirectoryPath = TestUtils.CreateTestDirectory(this.TestContext);
-            string testAssemblyPath = Path.Combine(testDirectoryPath, Path.GetFileName(noAnalyzerAssemblyPath));
-            File.Copy(noAnalyzerAssemblyPath, testAssemblyPath);
+            string testDirectoryPath = Path.Combine(TestContext.DeploymentDirectory, "oneAnalyzer");
+
+            string exampleAnalyzer1Path = Path.Combine(testDirectoryPath, "ExampleAnalyzer1.dll");
+            Assert.IsTrue(File.Exists(exampleAnalyzer1Path), "Test setup error: expected assembly does not exist: {0}", exampleAnalyzer1Path);
 
             // Act
             IEnumerable<DiagnosticAnalyzer> csharpDiagnostics = scanner.InstantiateDiagnostics(testDirectoryPath, LanguageNames.CSharp);
             IEnumerable<DiagnosticAnalyzer> vbDiagnostics = scanner.InstantiateDiagnostics(testDirectoryPath, LanguageNames.VisualBasic);
 
             // Assert
-            Assert.AreEqual(0, csharpDiagnostics.Count(), "No analyzers should have been detected");
-            Assert.AreEqual(0, vbDiagnostics.Count(), "No analyzers should have been detected");
+            // ConfigurableAnalyzer is both C# and VB, so should appear in both
+            Assert.AreEqual(2, csharpDiagnostics.Count(), "Expecting 2 C# analyzers");
+            Assert.AreEqual(2, vbDiagnostics.Count(), "Expecting 2 VB analyzers");
+
+            // Using name comparison because type comparison fails if the types are from assemblies with different paths (even if copied)
+            // Loaded from ExampleAnalyzer1.dll
+            Assert.IsNotNull(
+                csharpDiagnostics.SingleOrDefault(d => d.GetType().FullName == typeof(SimpleAnalyzer).FullName),
+                "Expected a SimpleAnalyzer");
+
+            Assert.IsNotNull(
+                csharpDiagnostics.SingleOrDefault(d => d.GetType().FullName == typeof(ConfigurableAnalyzer).FullName),
+                "Expected a ConfigurableAnalyzer");
+            
+            Assert.IsNotNull(
+                vbDiagnostics.SingleOrDefault(d => d.GetType().FullName == typeof(VBAnalyzer).FullName),
+                "Expected a VBAnalyzer");
+
+            Assert.IsNotNull(
+                vbDiagnostics.SingleOrDefault(d => d.GetType().FullName == typeof(ConfigurableAnalyzer).FullName),
+                "Expected a ConfigurableAnalyzer");
+
+            Assert.IsNull(
+                vbDiagnostics.SingleOrDefault(d => d.GetType().FullName == typeof(AbstractAnalyzer).FullName),
+                "Expected no abstract analyzers");
```

Comment:
As discussed in an email thread, it would be better not to reference the test analyser types directly i.e. using "*namespace*.SimpleAnalyzer" instead.